src/fem/patch_fe_values.hh

```diff
@@ -22,20 +22,148 @@
 #define PATCH_FE_VALUES_HH_
 
 
-//#include <string.h>                           // for memcpy
-//#include <algorithm>                          // for swap
-//#include <new>                                // for operator new[]
-//#include <string>                             // for operator<<
-//#include <vector>                             // for vector
-//#include "fem/element_values.hh"              // for ElementValues
-//#include "fem/fe_values.hh"                   // for FEValuesBase
-//#include "fem/fe_values_views.hh"             // for FEValuesViews
-//#include "mesh/ref_element.hh"                // for RefElement
-//#include "mesh/accessors.hh"
-//#include "fem/update_flags.hh"                // for UpdateFlags
-//#include "quadrature/quadrature_lib.hh"
-
-
-
-//
+#include <string.h>                           // for memcpy
+#include <algorithm>                          // for swap
+#include <new>                                // for operator new[]
+#include <string>                             // for operator<<
+#include <vector>                             // for vector
+#include "fem/element_values.hh"              // for ElementValues
+#include "fem/fe_values.hh"                   // for FEValuesBase
+#include "fem/fe_values_views.hh"             // for FEValuesViews
+#include "mesh/ref_element.hh"                // for RefElement
+#include "mesh/accessors.hh"
+#include "fem/update_flags.hh"                // for UpdateFlags
+#include "quadrature/quadrature_lib.hh"
+#include "fields/eval_subset.hh"
+
+template<unsigned int spacedim> class PatchFEValues;
+
+
+
+using Scalar = double;
+using Vector = arma::vec3;
+using Tensor = arma::mat33;
+
+template <class ValueType>
+class ElQ {
+public:
+    /// Constructor
+    ElQ()
```

Comment:
Default constructor zakázat. Accessor musí být funkční.

Reply:
Object deklarujeme jako data member v assemblacích a inicializaci provádíme až v metodě initialize. Pokud zakážeme default constructor musí být data member deklarován jako pointer na object, např. `ElQ<Scalar> *JxW_;` Je to možné?

Reply:
Nejlepší by bylo, tyto accessory inicializovat již v initializer listu konstruktoru Assembly<dim>. Pokud bude PatchFEValues předávaný jako parametr konstruktoru Assembly, tak by to mělo být možné. 
Pokud by s tím byly potíe, tak alternativou je Default konstruktor + assigment opertor + ten správný konstruktor. Zbytečné složitosti.